### DISCUSSION TITLE
More control in update settings

### DISCUSSION THREAD

**Discussion by ryzam** [Budibase/budibase#3312] [Ideas]
Hi budibase , I would suggest update setting in configuration conditions to have more control settings like set hide/show by default without checking value condition.

**Comment by mjashanks** [comment]
I know what you mean. The solution would be to have a `Hide` setting in the settings panel. However, our goal is to keep this panel as simple as possible - we are trying to avoid putting too much in there!

The tradeoff is that Show/Hide can only be controlled via conditions - which is actually the only circumstance where we need to hide a component.

It's a little bit annoying to default something to Hidden - but I think it's worth it, to keep the design panel clean.

**Reply by ryzam** [reply]
How about if I wanted to change container background color based on the databinding value for example in repeater?

**Reply by mjashanks** [reply]
Ah right - I may have misunderstood the first time. 

Is your issue that you cannot currently set the color in conditions? I can see that this is an issue with containers. Setting a color shouldn't be a problem, as we currently allow setting the text color for Headline components (in conditions)

We are definitely open to adding more in here. It would be good to get a list of settings that we are missing, so we can do them all in one pass.

**Reply by ryzam** [reply]
Something similar to this where the background color of container and button based on the value of http verb 

![image](https://user-images.githubusercontent.com/597759/141865015-61b9aa9f-c444-42e5-8653-bc96b7e49c69.png)

**Reply by ryzam** [reply]
I have another request, can budibase support update state in update setting of container. There is requirement where I need to generate random number and set to state to be used in the form, this should be handle on the background process when page load.
